src/profile-logic/profile-data.js

```diff
@@ -1165,21 +1171,31 @@ export function filterCounterToRange(
  * accumulatedCounts array.
  */
 export function accumulateCounterSamples(
-  samples: CounterSamplesTable
-): AccumulatedCounterSamples {
-  let minCount = 0;
-  let maxCount = 0;
-  let accumulated = 0;
-  const accumulatedCounts = [];
-  for (let i = 0; i < samples.length; i++) {
-    accumulated += samples.count[i];
-    minCount = Math.min(accumulated, minCount);
-    maxCount = Math.max(accumulated, maxCount);
-    accumulatedCounts[i] = accumulated;
+  samplesArray: Array<CounterSamplesTable>
+): Array<AccumulatedCounterSamples> {
```

Comment:
This function and other functions related to returns an array instead now. Since counter_groups are array now.